crates/kornia-apriltag/src/quad.rs

```diff
@@ -0,0 +1,1059 @@
+pub fn compute_line_fit_prefix_sums<A: ImageAllocator>(
+    src: &Image<Pixel, 1, A>,
+    gradient_infos: &[GradientInfo],
+) -> Vec<LineFit> {
+    let src_slice = src.as_slice();
+    // TODO: Find a way to avoid allocation
+    let mut lfps = vec![LineFit::default(); gradient_infos.len()];
+
+    gradient_infos.iter().enumerate().for_each(|(i, cluster)| {
+        if i > 0 {
+            lfps[i] = lfps[i - 1].clone();
+        }
+
+        let delta = 0.5f32;
+        let x = cluster.pos.x as f32 * 0.5 + delta;
+        let y = cluster.pos.y as f32 * 0.5 + delta;
+        let ix = x as usize;
+        let iy = y as usize;
+        let mut w = 1.0f32;
+
+        if ix > 0 && ix + 1 < src.width() && iy > 0 && iy + 1 < src.height() {
+            let grad_x = src_slice[iy * src.width() + ix + 1] as i32
+                - src_slice[iy * src.width() + ix - 1] as i32;
+
+            let grad_y = src_slice[(iy + 1) * src.width() + ix] as i32
+                - src_slice[(iy - 1) * src.width() + ix] as i32;
+
+            w = ((grad_x * grad_x + grad_y * grad_y) as f32).sqrt() + 1.0;
+        }
+
+        let fx = x;
+        let fy = y;
+
+        lfps[i].mx += w * fx;
+        lfps[i].my += w * fy;
+        lfps[i].mxx += w * fx * fx;
+        lfps[i].mxy += w * fx * fy;
+        lfps[i].myy += w * fy * fy;
+        lfps[i].w += w;
+    });
+
+    lfps
+}
+
+/// Segments the gradient information into four maxima corresponding to the corners of a quadrilateral.
+///
+/// This function analyzes the error profile of line fits over the gradient information and finds
+/// four maxima (peaks) in the error signal, which are interpreted as the corners of a quad. It
+/// returns `true` if a valid segmentation is found and writes the indices of the maxima into `indices`.
+///
+/// # Arguments
+///
+/// * `gradient_infos` - Slice of `GradientInfo` representing the cluster.
+/// * `lfps` - Slice of `LineFit` prefix sums for the cluster.
+/// * `indices` - Mutable reference to an array where the four corner indices will be written.
+/// * `opts` - Options for quad fitting process
+///
+/// # Returns
+///
+/// `true` if four valid maxima are found and written to `indices`, `false` otherwise.
+pub fn quad_segment_maxima(
+    gradient_infos: &[GradientInfo],
+    lfps: &[LineFit],
+    indices: &mut [usize; 4],
+    opts: FitQuadOpts,
+) -> bool {
+    // TODO: check if the length of gradient_infos and lfps is same
+    let len = gradient_infos.len();
+    let window_size = 20.min(len / 12);
+
+    // can't fit a quad, too few points
+    if window_size < 2 {
+        return false;
+    }
+
+    let mut errors = vec![0.0f32; len];
+
+    (0..len).for_each(|i| {
+        fit_line(
+            lfps,
+            (i + len - window_size) % len,
+            (i + window_size) % len,
+            None,
+            errors.get_mut(i),
+            None,
+        );
+    });
+
+    const SIGMA: f32 = 1.0;
```

Comment:
Can you check if we can use the guassian kernel defined here: 
https://github.com/kornia/kornia-rs/blob/171965ad2f41042c49ad21f904eb41afdda6e3c6/crates/kornia-imgproc/src/filter/kernels.rs#L25